extensions/spotify-player/src/components/NowPlayingDetail.tsx

```diff
@@ -0,0 +1,49 @@
+import { Icon, List } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { SpotifyPlayingState, SpotifyState, TrackInfo } from "../spotify/types";
+import { msToHMS } from "../utils";
+
+export default function NowPlayingDetail({
+  trackInfo,
+  playerState,
+}: {
+  trackInfo: TrackInfo;
+  playerState: SpotifyState;
+}): JSX.Element {
+  const [position, setPosition] = useState(playerState.position * 1000);
+  const trackDuration = trackInfo.duration;
+
+  useEffect(() => {
+    setPosition(playerState.position * 1000);
+  }, [playerState.position]);
+
+  useEffect(() => {
```

Comment:
This code is not ideal (given that its inherently inaccurate) but I think it provides a better experience in giving the user a close to realtime view of the progress which is updated whenever we fetch the data from the actual Spotify client.